### PR TITLE
Send auth failures to Datadog

### PR DESCRIPTION
**Pull request by rhymes**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Thanks to https://github.com/thepracticaldev/dev.to/issues/7134 I found out the failure path for the sign in process is broken as Timber is not enabled in development anymore. As logging the error in production logs might not actually be beneficial, I moved it to Datadog.

I got the options from https://docs.datadoghq.com/developers/metrics/dogstatsd_metrics_submission/

## Related Tickets & Documents

Helping to debug https://github.com/thepracticaldev/dev.to/issues/7134

## Added tests?

- [ ] yes
- [x] no, because they aren't needed
- [ ] no, because I need help

### CODE REVIEW

**Comment by rhymes** [comment]
> I was looking in Timber and we get these pretty regularly. I think rather than send them to Honeybadger where we would expect to have to act on them, we should send them to Datadog and then we can alert if we see them in excessive volume. What do you think?

Great idea! I'm going to change it :)

File: app/controllers/omniauth_callbacks_controller.rb [hunk 1]
```diff
@@ -15,19 +15,18 @@ def github
...
-        error_reason: error&.error_reason,
-        error_type: error&.error,
-        error_uri: error&.error_uri,
-        provider: request.env["omniauth.strategy"].name,
-        origin: request.env["omniauth.strategy.origin"],
-        params: request.env["omniauth.params"],
-        cookie: request.env["rack.request.cookie_hash"]
-      },
+    DatadogStatsClient.increment(
+      "auth.failure",
```
Comment (mstruve) [inline app/controllers/omniauth_callbacks_controller.rb:19]: Not blocking but should we name this metric `omniauth.failure` so if we end up sending metrics for other types of authentication we can differentiate?

File: app/controllers/omniauth_callbacks_controller.rb [hunk 2]
```diff
@@ -13,14 +13,22 @@ def github
...
-                         auth: request.env["omniauth.auth"],
-                         provider: request.env["omniauth.strategy"].to_s,
-                         cookie: request.env["rack.request.cookie_hash"]
-                       }
+    error = request.env["omniauth.error"]
+
+    DatadogStatsClient.increment(
+      "omniauth.failure",
+      tags: [
+        "class:#{error}",
```
Comment (citizen428) [inline app/controllers/omniauth_callbacks_controller.rb:21]: Are we sure this shouldn't be `error.class` or something?

Comment (rhymes) [reply app/controllers/omniauth_callbacks_controller.rb:21]: `error.to_s` is the same as `error.class.name` in this case so I kept it like that 🤷‍♂

Comment (citizen428) [reply app/controllers/omniauth_callbacks_controller.rb:21]: No worries @rhymes, just double checking.

**Pull request merged**